code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -159,10 +199,10 @@ void setImage(Map<String, Object> inspect, Instance instance) {
         String imageUuid = (String)config.get(INSPECT_IMAGE);
 
         // Somewhat of a hack, but needed for testing against sim contexts
-        if ( !imageUuid.matches(IMAGE_KIND_PATTERN) ) {
+        if (!imageUuid.matches(IMAGE_KIND_PATTERN)) {
             imageUuid = IMAGE_PREFIX + imageUuid;
         }
-        DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_IMAGE_UUID).set(imageUuid);
+        DataAccessor.fields(instance).withKey(FIELD_IMAGE_UUID).set(imageUuid);
         Image image = null;
         try {
             image = storageService.registerRemoteImage(imageUuid);
```

Comment:
io.cattle.platform.storage.service.StorageService.registerRemoteImage(String) is problematic.  I can't imagine the account_id is being set properly.  Also the lookup of the image will most likely cross accounts too.  This isn't specific to native docker but we need to fix.  registerRemoteImage() should really take an account id so that look ups are scoped and the proper account id gets set.

When registerRemoteImage() is ran during an API call, the account_id does get set properly because there is a thread local context variable that has the account_id in it.